### DISCUSSION TITLE
Can't get version 0.7.0 compile on Xcode 14 beta 1

### DISCUSSION THREAD

**Discussion by xiaogdgenuine** [pointfreeco/combine-schedulers#61] [Q&A] [answered]
**Describe the bug**
Thank you for the concurrency work!

Hoever I got a compile error when try to use latest TCA package `0.39.0`, which depends on this package at version `0.7.0`.

`AnyScheduler.swift:333:48: error build: Protocol type with type arguments can only be used as a generic constraint`

**To Reproduce**
Download the package, and compile.

**Expected behavior**
It should compile.

**Screenshots**
<img width="1758" alt="image" src="https://user-images.githubusercontent.com/51731239/183817615-5eb317e9-066a-45cc-862c-b6bb3a8a027c.png">

**Environment**
 - Xcode 14.0 beta 1
 - Swift 5.7
 - OS iOS 16 beta(simulator)
 - macOS 12.4 Monterey

**Comment by mbrandonw** [comment, marked as answer]
Hi @xiaogdgenuine, the primary associated type for `Scheduler` wasn't added until a later beta, and I don't believe it's possible for us to conditionally compile code based on when that feature became available.

In general we can only really support the newest Xcode beta. Is it possible for you to use a newer beta?

Also, since this isn't really an issue with the library I am going to convert it to a discussion.

**Reply by xiaogdgenuine** [reply]
Oh I see, that's new for me too, then this is totally make sense, I will use a newer beta, thanak you!
